### PR TITLE
Migrate Fabric component to function component

### PR DESCRIPTION
**Pull request by MLoughry**
#### Pull request checklist

- [x] Addresses an existing issue: #12770
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Migrate the `Fabric` component to a function component

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12923)

### CODE REVIEW

File: packages/react-next/src/components/Fabric/Fabric.base.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    // Or If contextDir !== pageDir
+    // Then we need to set dir of the Fabric root
+    rootDir: componentDir !== contextDir || componentDir !== pageDir ? componentDir : dir,
+    // If dir !== contextDir || pageDir
+    // then set contextual theme around content
+    needsTheme: componentDir !== contextDir,
+  };
+};
+
+// tslint:disable-next-line:function-name no-function-expression
```
Comment (xugao) [inline packages/react-next/src/components/Fabric/Fabric.base.tsx:36]: should be anonymous function + displayName ?

File: packages/react-next/src/components/Fabric/Fabric.base.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    if (applyThemeToBody) {
+      const currentDoc = getDocument(rootElement.current);
+      if (currentDoc) {
+        currentDoc.body.classList.add(bodyThemed);
+        return () => {
+          currentDoc.body.classList.remove(bodyThemed);
+        };
+      }
+    }
+  }, [bodyThemed]);
```
Comment (xugao) [inline packages/react-next/src/components/Fabric/Fabric.base.tsx:91]: i think we need `rootElement.current` here too

Comment (MLoughry) [reply packages/react-next/src/components/Fabric/Fabric.base.tsx:91]: A change to the ref value doesn't take effect until the render is pushed to the DOM, won't trigger a rerender. If we're concerned about it, we'd have to use a state value instead.

Comment (xugao) [reply packages/react-next/src/components/Fabric/Fabric.base.tsx:91]: makes sense

**Pull request merged**